### ISSUE TITLE
[Bug]: could not load library "C:/Program Files/PostgreSQL/12/lib/timescaledb-2.11.2.dll"

### ISSUE THREAD

**Issue by Isildur981** [timescale/timescaledb#6055]
### What type of bug is this?

Configuration

### What subsystems and features are affected?

Configuration

### What happened?

Altering the extension results in ERROR:  could not load library "C:/Program Files/PostgreSQL/12/lib/timescaledb-2.11.2.dll": The specified module could not be found.

The "ALTER EXTENSION timescaledb UPDATE;" command was run from psql with the -X flag.

I am currently running the Timescale 2.11.1 extension and was attempting to update to 2.11.2.

### TimescaleDB version affected

2.11.2

### PostgreSQL version used

12.16

### What operating system did you use?

Windows Server 2016 Standard x64

### What installation method did you use?

Other

### What platform did you run on?

On prem/Self-hosted

### How can we reproduce the bug?

```bash
Have PostgreSQL Server version 12.16 running on Windows with the 2.11.1 Timescale extension installed and attempt to update the Timescale extension to 2.11.2.
```

**Comment by jnidzwetzki** [comment]
Hello @Isildur981,

I was able to reproduce the problem in my local environment. It happens when OpenSSL 3.x is not installed. Since TimescaleDB 2.11.2, OpenSSL 3.x is required on Windows. After you have installed this version of OpenSSL, you should be able to install TimescaleDB 2.11.2.

**Comment by Isildur981** [comment]
Jan,

I didn't compile PostgreSQL.  I downloaded the installer hosted by EDB that
the official Postgres web site links to.  I don't know if OpenSSL 1.1 is
installed or not.  The Timescale documentation only mentions an OpenSSL
requirement for Postgres version 15.1.1 or later.  Nothing is said about
requiring Open SSL for versions 12, 13, or 14.

Are there instructions in the documentation for how to install the required
OpenSSL version?

Forrest Lanier

**Comment by jnidzwetzki** [comment]
Hello @Isildur981,

Thanks for getting back. The [documentation](https://docs.timescale.com/self-hosted/latest/install/installation-windows/#prerequisites) for installing TimescaleDB on Windows was updated recently. Depending on the used TimescaleDB version, you need to install OpenSSL 1.1 or 3 on your system. On [this page](https://wiki.openssl.org/index.php/Binaries), you will find an overview of the available OpenSSL builds for Windows.

**Comment by Isildur981** [comment]
Jan,

Thanks for the information.

Forrest

**Comment by jnidzwetzki** [comment]
Hello @Isildur981,

Thanks for getting back. Has the installation of OpenSSL solved the issue? Can we close this ticket or is there anything else we can assist you with?

**Comment by Isildur981** [comment]
Jan,

I haven't been able to try installing OpenSSL 3.0.  The server is used in
production, and I can't install new software on it until we have some
downtime, and that might not happen for a couple of days
to several weeks.  You can go ahead and close the ticket if you need to.

Forrest

**Comment by Isildur981** [comment]
Jan,

I just wanted to follow up and let you know that we had some downtime this
past weekend, and I was able to install OpenSSL 3.0 and successfully update
the Timescale extension.  Thanks for your help again.

Forrest


---------- Forwarded message ---------
From: Forrest Lanier ***@***.***>
Date: Thu, Sep 14, 2023 at 4:33 PM
Subject: Re: [timescale/timescaledb] [Bug]: could not load library
"C:/Program Files/PostgreSQL/12/lib/timescaledb-2.11.2.dll" (Issue #6055)
To: timescale/timescaledb <
***@***.***>


Jan,

I haven't been able to try installing OpenSSL 3.0.  The server is used in
production, and I can't install new software on it until we have some
downtime, and that might not happen for a couple of days
to several weeks.  You can go ahead and close the ticket if you need to.

Forrest

**Comment by jnidzwetzki** [comment]
Hello @Isildur981,

Thanks for the update. Great to hear that the problem has been fixed with the installation of OpenSSL 3.0.
